src/PushCommand.php

```diff
@@ -400,6 +412,56 @@ private function getPackageName(InputInterface $input = null)
         }
     }
 
+    /**
+     * @param InputInterface $input
+     */
+    private function parseNexusExtra(InputInterface $input){
+        try{
+            $this->checkNexusPushValid($input);
+
+            $repository = $input->getOption(self::REPOSITORY);
+            $extras = $this->getComposer(true)->getPackage()->getExtra();
+            if(empty($repository)){
+                // configurations in composer.json support Only upload to unique repository
+                if(!empty($extras['nexus-push'])){
+                    $this->nexusPushConfig = $extras['nexus-push'];
+                }
+
+            }else{
+                // configurations in composer.json support upload to multi repository
+                foreach ($extras['nexus-push'] as $key=> $nexusPushConfigItem){
+                    if(empty($nexusPushConfigItem[self::PUSH_CFG_NAME])){
+                        $fmt = 'The nexus-push configuration array in composer.json with index {%s} need provide value for key "%s"';
+                        $exceptionMsg = sprintf($fmt,$key,self::PUSH_CFG_NAME);
+                        throw new InvalidConfigException($exceptionMsg);
+                    }
+                    if($nexusPushConfigItem[self::PUSH_CFG_NAME] ==$repository){
+                        $this->nexusPushConfig = $nexusPushConfigItem;
+                    }
+                }
+
+                if(empty($this->nexusPushConfig)){
+                    throw new InvalidArgumentException('The value of option --repository match no nexus-push configuration, pleash check');
+                }
+            }
+        }catch (\Exception $e){
+            $this->getIO()
+                ->write($e->getMessage());
+            throw $e;
+        }
+    }
+
+    public function checkNexusPushValid(InputInterface $input){
```

Comment:
I don't think it's relevant to put this function `public` (if we put it public, users might think it's part of the public API and 
that they can use it, and I think it should currently stay internal).